install/kubernetes/cilium/templates/cilium-operator/deployment.yaml

```diff
@@ -277,6 +286,15 @@ spec:
       nodeSelector:
         {{- toYaml . | trim | nindent 8 }}
       {{- end }}
+      {{- if and .Values.clustermesh.config.enabled (not (and .Values.clustermesh.useAPIServer .Values.clustermesh.apiserver.kvstoremesh.enabled )) }}
```

Comment:
Let's guard it with `enableEndpointSliceSynchronization` just in case so it will be aligned with other changes and we won't change default behavior.